File: src/main/java/org/chousik/commands/ShowCommand.java
package org.chousik.commands;

import org.chousik.exception.ArgumentCountError;
import org.chousik.handlers.ICollectionController;

/**
 * Команда show. Позволяет вывести коллекцию.
 */
public class ShowCommand extends ACommand {
    private ICollectionController collectionController;

    public ShowCommand(ICollectionController collectionController) {
        super("show", " команда позволяет вывести коллекцию.", 0);
        this.collectionController = collectionController;
    }

    /**
     * Метод для вывода коллекции
     *
     * @param args аргументы
     * @throws ArgumentCountError если количество аргументов не совпадает
     */
    @Override
    public void execute(String[] args) throws ArgumentCountError {
        valideCountsArgument(args);
        System.out.println(collectionController.getElements());
    }
}

File: src/main/kotlin/commands/ShowCommand.kt
package commands

import org.chousik.handlers.ICollectionController


class ShowCommand(private val collectionController: ICollectionController<*>) :
    ACommand("show", " команда позволяет вывести коллекцию.", 0) {
    override fun execute(args: Array<String>) {
        validCountsArgument(args)
        println(collectionController.elements)
    }
}